store.go

```diff
@@ -62,30 +58,26 @@ func NewStore(dbFile string) (Store, error) {
 		collections: map[string]*collection{},
 	}
 
-	st.globalCollection = &collection{
-		Name:    GlobalCollection,
-		db:      st.db,
-		refMake: defaultReferenceCreator,
-	}
-	st.globalCollection.globalCollection = st.globalCollection
 	return st, nil
 }
 
 func (s *store) Collection(name string) Collection {
-	if name == GlobalCollection {
-		return s.globalCollection
-	}
-
 	c, ok := s.collections[name]
 	if !ok {
 		c = &collection{
-			Name:             name,
-			db:               s.db,
-			globalCollection: s.globalCollection,
-			refMake:          defaultReferenceCreator,
+			Name:    name,
+			db:      s.db,
+			refMake: defaultReferenceCreator,
```

Comment:
since there's only 1 `referenceCreator` (right?) wouldn't be it easier to drop the possible error return from the signature, since it'll probably never be implemented another time _with_ error return? (since it'll always be a hash)

Reply:
even more, since you can't create a Store yet with your own....... removed error from sig